tests/test_session_key.py

```diff
@@ -264,3 +275,42 @@ async def test_dapp_long_signature(starknet: Starknet, contracts):
     )
     # check it worked
     assert (await dapp.get_balance().call()).result.res == 47
+
+@pytest.mark.asyncio
+async def test_executeOnPlugin(starknet: Starknet, contracts):
+    # Account 2 tries to revoke a session key on Account 1, via executeOnPlugin and via readOnPlugin
+
+    account, stark_plugin_signer, stark_plugin_signer_2, session_plugin_signer, dapp1, dapp2, session_key_class = contracts
+
+    await stark_plugin_signer.add_plugin(session_key_class)
+    update_starknet_block(starknet=starknet, block_timestamp=DEFAULT_TIMESTAMP)
+
+    session = build_session(
+        signer=stark_plugin_signer,
+        allowed_calls=[(dapp1.contract_address, 'set_balance')],
+        session_public_key=session_key.public_key,
+        session_expiration=DEFAULT_TIMESTAMP + 10,
+        chain_id=StarknetChainId.TESTNET.value,
+        account_address=account.contract_address
+    )
+
+    revoke_session_key_arguments = [session.session_hash]
```

Comment:
it should be changed with the public key 

```suggestion
    revoke_session_key_arguments = [session.session_public_key]
```

Reply:
hi, i think you are not right this time. Looking at the plugin code, it's the hash that we revoke. I think it makes sense since the hash includes the public key and the other params. Technically you can have many sessions with the same public key

Reply:
https://github.com/argentlabs/starknet-plugin-account/blob/367181f26b73830b8ea87a7d7dd8bee1ed713b11/contracts/plugins/SessionKey.cairo#L96

https://github.com/argentlabs/starknet-plugin-account/blob/367181f26b73830b8ea87a7d7dd8bee1ed713b11/contracts/plugins/SessionKey.cairo#L133

Reply:
you are right @CremaFR , it's checking the session key. Should we change it to check the session_hash instead?

Reply:
anyways, i changed it in this PR. If we want to change the plugin behaviour we can do it separately

Reply:
what did you change? the test in the current version is not revoking the session key. note that is does matter since it's only to check reverts anyway

Reply:
i forgot to commit the change 🤦  pushed now